modules/geo-layers/src/tile-layer/tileset-2d.js

```diff
@@ -269,11 +278,22 @@ export default class Tileset2D {
   }
   /* eslint-enable complexity */
 
+  _abortUnselectedTileRequests() {
+    for (const tile of this._cache.values()) {
+      // Abort ongoing requests for tiles that are no longer visible
+      // TODO: Make this based on max outstanding requests rather than immediately aborting off-screen requests
+      if (!tile.isVisible && !tile.isLoaded) {
```

Comment:
I think you want to use `isSelected`, not `isVisible`. The latter is true even if the zoom level has changed dramatically. `isSelected` is true only when the tile is in view and is at the current zoom level.